app/controllers/api/v1/users/registrations_controller.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    module Users
+      class RegistrationsController < Devise::RegistrationsController
+        respond_to :json
+
+        before_action :ensure_valid_client
+
+        def create
+          user = User.from_email(user_params[:email], user_params[:password])
+          user.errors.empty? ? render_success(user) : render_error(user)
+        end
+
+        private
+
+        def user_params
+          params.require(:user).permit(:email, :password)
+        end
+
+        def render_success(user)
+          data = UserSerializer.new(user).serializable_hash[:data]
+          message = I18n.t('activemodel.notices.models.user.create')
+
+          render json: { data: data, meta: { message: message } }, status: :created
```

Comment:
Thank you, I'll keep the message so it's aligned with `search_results_controller.rb`

In terms of `serializable_hash`, I think it's necessary if we only want the fields inside `data`

Otherwise it will look like this (we don't want `data` inside `data`):

![Screen Shot 2023-07-11 at 14 09 36](https://github.com/nimblehq/ic-rails-huey-toby/assets/18277915/415cfffe-e74b-44bc-bcea-ed0ad590b064)

Reply:
Then you can remove the root data attribute 😄 
```suggestion
          message = I18n.t('activemodel.notices.models.user.create')
          user_serializer = UserSerializer.new(user, meta: { message: message })

          render json:user_serializer, status: :created
```